src/fides/lib/models/client.py

```diff
@@ -104,6 +113,7 @@ def create_access_code_jwe(self, encryption_key: str) -> str:
             JWE_PAYLOAD_CLIENT_ID: self.id,
             JWE_PAYLOAD_SCOPES: self.scopes,
             JWE_ISSUED_AT: datetime.now().isoformat(),
+            JWE_PAYLOAD_ROLES: self.roles,
```

Comment:
This additionally adds the roles to the access token for the client.  We wait until `verify_oauth_util` to check to see if the token has scopes directly or a scope via a role in order to access the endpoint.  This way, if we scopes associated with a role change, the tokens are still valid.